definitions/divorce/json/CaseField.json

```diff
@@ -2906,5 +2898,29 @@
     "Label": "### Outcome of DN:",
     "FieldType": "Label",
     "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "DIVORCE",
+    "ID": "DecreeNisiGranted",
+    "Label": "Grant Decree Nisi?",
+    "FieldType": "YesOrNo",
+    "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "DIVORCE",
+    "ID": "CostsClaimGranted",
+    "Label": "Grant Cost Order?",
+    "FieldType": "YesOrNo",
+    "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "01/01/2017",
+    "CaseTypeID": "DIVORCE",
+    "ID": "state",
```

Comment:
You don't need the state as a CaseField.
https://github.com/hmcts/ccd-data-store-api/blob/master/src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java#L193

On return of a CCDCallbackResponse, as long as the key "state" is in the data, it will use that as the next state instead of the defined postState, then remove that key from the data to update CCD with. So even if "state" is a CCD field, it won't be populated via the callback.

Reply:
Nevermind, after more conversations looks like you can't progress without the field. Strange, because it would never be set following the code logic.

Reply:
Can we clarify this with CCD team?

Reply:
Yes I think this needs to be clarified first. I just realised if the state field has CRU permissions from LA, they can technically populate it with any event by accident (although its highly unlikely). If the state field is populated, whichever callback triggers next will move to the populated state incorrectly, since this field seems to only be cleared during the Callback Invocation.

Reply:
Hello guys. I've created the following tech debt to handle this once CCD apply this change:
https://tools.hmcts.net/jira/browse/DIV-4956